packages/hurl/src/http/client.rs

```diff
@@ -577,10 +581,15 @@ impl Client {
         }
 
         if let Some(user) = &options.user {
-            if options.aws_sigv4.is_some() || options.ntlm {
+            if options.aws_sigv4.is_some() || options.ntlm || options.negotiate {
```

Comment:
@ashishajr I've missed this line in the `--ntlm` PR, would you be kind to make a PR updating the comment and explaining why we take `options.ntlm` into account here? The comment is only refererring to `aws_sigv4` I don't see the link with `ntlm`

Reply:
Yes the comment is OK, it explains why we're doing it for ntlm, negotiate, awssigv4